contracts/origination/OriginationControllerSTIRFRY.sol

```diff
@@ -0,0 +1,268 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "./OriginationControllerBase.sol";
+
+import "../interfaces/IOriginationControllerSTIRFRY.sol";
+import "../interfaces/IFeeController.sol";
+import "../interfaces/IVaultFactory.sol";
+
+import "../libraries/Constants.sol";
+
+import {
+    OCS_ZeroAddress,
+    OCS_InvalidStirfryPair,
+    OCS_InvalidPrincipalAmounts,
+    OCS_InvalidVaultAmount,
+    OCS_InvalidInterestAmounts
+} from "../errors/Lending.sol";
+
+/**
+ * @title OriginationControllerSTIRFRY
+ * @author Non-Fungible Technologies, Inc.
+ *
+ * STIRFRY - Set Term Interest Rate Fixed Rate Yield
+ *
+ * This contract is similar to the vanilla OriginationController, with a few key changes:
+ * 1). The collateral is collected from the lender instead of the borrower.
+ * 2). The borrower does not receive any principal at the start of the loan. Instead, the borrower deposits the
+ *     interest amount into the lenders vault. This vault is the vault that is being used for collateral in the loan.
+ * 3). There are no loan rollovers.
+ *
+ * In a STIRFRY loan scenario, a lender will mint a vault and deposit into the vault an ERC20 that accumulates some
```

Comment:
Also want to challenge using the term "loan" here - the product itself is not a loan, it's a swap - the payoff curve for both counterparties looks different. I.e. the "borrower" (not really a borrower anymore) thinks of themselves as entering a "trade" as opposed to taking out a loan, and when they enter the trade they are not getting anything delivered to their wallet (not borrowing anything). For instance, there will be no "borrow" button on the product UI, it will be "fill order/trade". And there will be no "repay" button - it will be "settle/close position", which causes something that looks like repayment from LoanCore's perspective to happen, but that last part is just an implementation detail